test/e2e/framework/metrics_util.go

```diff
@@ -228,6 +229,77 @@ func (l *SchedulingMetrics) PrintJSON() string {
 	return PrettyPrintJSON(l)
 }
 
+type histogram struct {
+	Count   int
+	Buckets map[float64]int
+}
+
+func (h *histogram) ConvertToRangeBuckets() {
+	keys := []float64{}
+	for k := range h.Buckets {
+		keys = append(keys, k)
+	}
+	sort.Float64s(keys)
+	lessValuesSum := 0
+	for _, k := range keys {
+		h.Buckets[k] -= lessValuesSum
+		lessValuesSum += h.Buckets[k]
+	}
+}
+
+func (h *histogram) CalculatePercs() (float64, float64, float64) {
```

Comment:
Can you instead make this a function that takes single argument (like 0.5, 0.9, etc) and returns a single value for the percentile value?

Reply:
done